### ISSUE TITLE
NVDA does not read hints or error messages when tabbing to checkboxes

### ISSUE THREAD

**Issue by matteason** [alphagov/govuk-frontend#3625]
## Description of the issue

This is an NVDA bug, but I'm filing an issue here because it affects the accessibility of GOV.UK Frontend so I thought you may want a record of it.

When a checkbox inside a fieldset gets focus, any hint or error message associated with the fieldset should be read out after the legend and before the checkbox label. They're associated with the fieldset via `aria-describedby`. However, NVDA only reads the legend and checkbox label, not the hint or error. This means that NVDA users may be unaware that the grouping has a hint or an error.

For example, the speech output for [this example](https://design-system.service.gov.uk/components/checkboxes/error/index.html) with a hint and an error is:

> What is your nationality?  grouping  clickable  British  check box  not checked    including English, Scottish, Welsh and Northern Irish

Note that this only affects checkboxes. Radios use the same structure and are unaffected.

[An NVDA issue for this bug](https://github.com/nvaccess/nvda/issues/12718) has been open since 2021 but unfortunately it doesn't look like there's been much progress towards a fix. It looks like [this commenter](https://github.com/nvaccess/nvda/issues/12718#issuecomment-1533364655) has also come across this issue while using GOV.UK Frontend though.

## Steps to reproduce the issue

See CodePen at https://codepen.io/matteason/pen/PoyBwYM or one of the GOV.UK Design System examples (eg [checkboxes](https://design-system.service.gov.uk/components/checkboxes/with-none-option-in-error/index.html))

Open NVDA, navigate to the example page and tab through the form fields

## Actual vs expected behaviour

Hints should be read for all component types in the example, but they're only read for text inputs and radios

## Environment (where applicable)

- Operating system: Windows 10
- Browser: Firefox 112.0, Chrome 112.0.5615.138
- GOV.UK Frontend Version: 4.6.0
